app/components/signup-steps/step-one.js

```diff
@@ -1,15 +1,47 @@
 import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
 import { debounce } from '@ember/runloop';
 import { JOIN_DEBOUNCE_TIME } from '../../constants/join';
-
+import { APPS } from '../../constants/urls';
 export default class SignupStepsStepOneComponent extends Component {
+  @tracked data = { firstname: '', lastname: '' };
+  @tracked isValid = true;
+  @tracked username = '';
   @action inputHandler(e) {
     const { onChange } = this.args;
     const passVal = () => {
+      this.data = { ...this.data, [e.target.name]: e.target.value };
       onChange(e.target.name, e.target.value);
+      if (this.data.firstname.trim() > '' && this.data.lastname.trim() > '') {
+        this.isValid = false;
+      } else {
+        this.isValid = true;
+      }
     };
 
-    debounce(passVal, JOIN_DEBOUNCE_TIME);
+    debounce(this.data, passVal, JOIN_DEBOUNCE_TIME);
+  }
+
+  @action async getUsername() {
+    try {
+      const firstname = this.data.firstname.toLowerCase();
+      const lastname = this.data.lastname.toLowerCase();
```

Comment:
this needs to be handle `inputHandler` function, try making `getUsername` only for API request

Reply:
Done with changes can you please check my latest commit

https://github.com/Real-Dev-Squad/website-www/assets/107163260/cc9fc52f-67e8-4a9e-812a-bcd64eb3e67a